src/main/purgeDeletedSiteData.ts

```diff
@@ -0,0 +1,14 @@
+import * as LocalMain from '@getflywheel/local/main';
+import { SiteDataBySiteID } from '../types';
+
+export function purgeDeletedSiteData(existingSiteDataBySiteID: SiteDataBySiteID, serviceContainer: LocalMain.ServiceContainerServices) {
```

Comment:
Could we rename this function to something like `returnSiteDataIfExists` or something that is more accurate to what the function actually does?

Reply:
I think I get where that's coming from, but I do like the current name from an "API" standpoint. Even though it is technically only giving back existing Local site data, looking at the function as a black box, you pass it an object of SiteData and then gives back the same object (not really the same object, but loosely the same) minus any entries that are associated with non-existent sites.

Maybe an equivalent example with slightly different logic helps demonstrates this?

```ts
export function purgeDeletedSiteData(siteDataBySiteID: SiteDataBySiteID, serviceContainer: LocalMain.ServiceContainerServices) {
	for (const id of Object.keys(siteDataBySiteID)) {
		if (!serviceContainer.siteData.getSite(id)) {
			delete siteDataBySiteID[id];
		}
	}

	return siteDataBySiteID;
}

```

Anyways, not going to die on a hill about this, but I'm not in love with the name `returnSiteDataIfExists`. Maybe we can land on something that provides some better clarity from an API perspective.

Thoughts?

Reply:
I'm okay if we keep it as-is. It feels a little off to me since we're instantiating the store with it (so it feels odd to set up a store via a purge function,) but I see your point here and there's not a hard logical reason why we should change it.

Reply:
Circling back to this...

I'm going to rename this `filterDeletedSiteData` and add a tsDoc block. I think those two things should help with clarity here.